R/qualification-ddi.R

```diff
@@ -0,0 +1,344 @@
+    plotDDIDataFrame <- NULL
+    plotDDIMetadata <- list()
+    plot <- configurationPlan$plots$DDIRatioPlots[[plotNumber]]
+
+    plotDDIMetadata$title <- plot$Title
+    plotDDIMetadata$sectionID <- plot$SectionId
+    # Pipes in configuration plan will be deprecated moving forward
+
+    plotDDIMetadata$plotTypes <- plot$PlotTypes %||% ospsuite::toPathArray(plot$PlotType)
+    plotDDIMetadata$axesSettings <- lapply(plotDDIMetadata$plotTypes, function(pltType) {
+      getAxesSettings(configurationPlan$plots$AxesSettings[[ddiPlotAxesSettings[[pltType]]]])
+    })
+    names(plotDDIMetadata$axesSettings) <- plotDDIMetadata$plotTypes
+
+
+    plotDDIMetadata$groups <- list()
+
+    pkParameters <- NULL
+    if(!is.null(plot$PKParameter)){
+      pkParameters <- toPathArray(plot$PKParameter)
+    }
+
+    for (groupNumber in seq_along(plot$Groups)){
+      group <- plot$Groups[[groupNumber]]
+      plotDDIMetadata$groups[[groupNumber]] <- list()
+      plotDDIMetadata$groups[[groupNumber]]$caption <- group$Caption
+      plotDDIMetadata$groups[[groupNumber]]$color <- group$Color
+      plotDDIMetadata$groups[[groupNumber]]$symbol <- tlfShape(group$Symbol)
+
+      for (ddiRatio in group$DDIRatios){
+        outputPath <- ddiRatio$Output
+        observedDataSet <- ddiRatio$ObservedData
+        observedDataSetFilePath <- configurationPlan$getObservedDataPath(id = observedDataSet)
+        observedDataRecordId <- ddiRatio$ObservedDataRecordId
+        observedDataFrame <- readObservedDataFile(file = observedDataSetFilePath)
+        validateIsIncluded(observedDataRecordId, observedDataFrame$ID)
+
+        ratioList <- list()
+
+        for (pkParameter in  pkParameters){
+          ratioList[[pkParameter]] <- list()
+
+          validateIsIncluded(ddiPKRatioColumnName[[pkParameter]], names(observedDataFrame))
+          ratioList[[pkParameter]]$observedRatio <- observedDataFrame[[ddiPKRatioColumnName[[pkParameter]]]][observedDataFrame$ID == observedDataRecordId]
+
+          for (simulationType in c("SimulationControl","SimulationDDI")){
+
+            plotComponent <- ddiRatio[[simulationType]]
+            projectName <- plotComponent$Project
+            simulationName <- plotComponent$Simulation
+
+            startTime <-  ospsuite::toBaseUnit(quantityOrDimension = ospDimensions$Time,
+                                               values = plotComponent$StartTime,
+                                               unit = plotComponent$TimeUnit)
+
+            endTime <- ospsuite::toBaseUnit(quantityOrDimension = ospDimensions$Time,
+                                            values = plotComponent$EndTime,
+                                            unit = plotComponent$TimeUnit)
+
+            pkParameterName <- generateDDIPlotPKParameterName(pkParameter,startTime,endTime)
+
+            simulationResultsFile <- configurationPlan$getSimulationResultsPath(project = projectName,
```

Comment:
sometime project is on the same line, sometime on the next line... I don't really care which but let's be consistent